pybind_interface/pybind_main.cpp

```diff
@@ -40,7 +40,7 @@ template <typename T>
 T parseOptions(const py::dict &options, const char *key) {
   if (!options.contains(key)) {
     char msg[100];
-    std::sprintf(msg, "Argument %s is not provided.\n", key);
+    std::snprintf(msg, sizeof(msg), "Argument %s is not provided.\n", key);
```

Comment:
we should use C++ types if clang needs us to fix it here:
```
  std::string msg = std::string("Argument ") + key + " is not provided.\n";
```

Reply:
You're right. I'll make the change.